salt/kubernetes-minion/kubelet.jinja

```diff
@@ -23,8 +23,11 @@ KUBELET_ARGS="\
     --pod-manifest-path=/etc/kubernetes/manifests \
     --pod-infra-container-image={{ pillar['pod_infra_container_image'] }} \
 {% endif -%}
-{% if salt['pillar.get']('infrastructure', 'libvirt') == 'aws' -%}
-    --cloud-provider=aws \
+{% if salt['pillar.get']('cloud:provider', '') != '' -%}
+               --cloud-provider={{ pillar['cloud']['provider'] }} \
+  {% if salt['pillar.get']('cloud:provider', '') == 'openstack' -%}
```

Comment:
This could get messy in the future - vSphere also needs a `--cloud-config`. I don't know about Azure. For now, this is fine, but we might want to make it more generic later.